frame/society/src/lib.rs

```diff
@@ -0,0 +1,1372 @@
+		/// # </weight>
+		///
+
+		#[weight = SimpleDispatchInfo::FixedNormal(30_000)]
+		pub fn payout(origin) {
+			let who = ensure_signed(origin)?;
+
+			ensure!(Self::is_member(&who), Error::<T, I>::NotMember);
+
+			let mut payouts = <Payouts<T, I>>::get(&who);
+			if let Some((when, amount)) = payouts.first() {
+				if when <= &<system::Module<T>>::block_number() {
+					T::Currency::transfer(&Self::payouts(), &who, *amount, KeepAlive)?;
+					payouts.remove(0);
+					if payouts.is_empty() {
+						<Payouts<T, I>>::remove(&who);
+					} else {
+						<Payouts<T, I>>::insert(&who, payouts);
+					}
+					return Ok(())
+				}
+			}
+			Err(Error::<T, I>::NoPayout)?
+		}
+
+		/// Found the society.
+		///
+		/// This is done as a discrete action in order to allow for the
+		/// module to be included into a running chain.
+		///
+		/// The dispatch origin for this call must be from the _FounderOrigin_.
+		///
+		/// Parameters:
+		///
+		/// - `founder` - The first member and head of the newly founded society.
+		///
+		/// # <weight>
+		/// - One storage read to check `Head`.
+		/// - One storage write to add the first member to society.
+		/// - One storage write to add new Head.
+		/// - One event.
+		///
+		/// This is marked as `Operational` since it is a low level system action.
+		/// # </weight>
+		///
+
+		#[weight = SimpleDispatchInfo::FixedOperational(10_000)]
+		fn found(origin, founder: T::AccountId) {
+			T::FounderOrigin::ensure_origin(origin)?;
+			ensure!(!<Head<T, I>>::exists(), Error::<T, I>::AlreadyFounded);
+			Self::add_member(&founder)?;
+			<Head<T, I>>::put(&founder);
+			Self::deposit_event(RawEvent::Founded(founder));
+		}
+
+		/// Allow founder origin to make judgement on a suspended member.
+		///
+		/// If a suspended member is forgiven, we simply add them back as a member, not affecting
+		/// any of the existing storage items for that member.
+		///
+		/// If a suspended member is rejected, remove all associated storage items, including
+		/// their payouts, and remove any vouched bids they currently have.
+		///
+		/// The dispatch origin for this call must be from the _SuspensionJudgementOrigin_.
+		///
+		/// Parameters:
+		///
+		/// - `who` - The suspended member to be judged.
+		/// - `forgive` - A boolean representing whether the suspension judgement origin
+		///               forgives (`true`) or rejects (`false`) a suspended member.
+		///
+		/// # <weight>
+		/// - One storage read to check `who` is a suspended member.
+		/// - Up to one storage write with O(log M) binary search to add a member back to society.
+		/// - Up to 3 storage removals to clean up a removed member.
+		/// - Up to one storage write with O(B) search to remove vouched bid from bids.
+		/// - Up to one additional event if unvouch takes place.
+		/// - One storage removal.
+		/// - One event for the judgement.
+		///
+		/// This is marked as `Operational` since it comes from a privileged origin.
```

Comment:
Incorrect. None of this module is even remotely security-critical to the network.